File: modele_classes_java_simple/Line.java
import java.util.List;

/**
 * Ligne téléphonique
 */
public class Line {
    
    // Attributs
    private String id;
    private String msisdn;
    private String utilisateur;
    private double forfait;
    private CycleFacturation cycle;
    private String statut;
    
    // Relations
    private Company company;
    private User employe;
    private List<Invoice> invoices;
    private List<Cycle> cycles;
}

File: modele_classes_csharp/Line.cs
using System.Collections.Generic;
using Moov.Facturation.Utilisateurs;
using Moov.Facturation.Facturation;
using Moov.Facturation.Catalogue;

namespace Moov.Facturation.Contrats
{
    /// <summary>
    /// Ligne téléphonique d'un contrat
    /// </summary>
    public class Line
    {
        // Attributs
        public string Id { get; set; }
        public string Msisdn { get; set; } // unique (numéro 8 chiffres)
        public string Utilisateur { get; set; }
        public double Forfait { get; set; }
        public CycleFacturation Cycle { get; set; }
        public string Statut { get; set; }
        
        // Relations
        public Company Company { get; set; }
        public User Employe { get; set; }
        public List<Invoice> Invoices { get; set; }
        public List<Cycle> Cycles { get; set; }
        
        // Constructeur
        public Line()
        {
            Invoices = new List<Invoice>();
            Cycles = new List<Cycle>();
        }
    }
}
